pkg/controller/master-controller-manager/rbac/sync_project.go

```diff
@@ -108,34 +108,53 @@ func (c *projectController) ensureProjectIsInActivePhase(ctx context.Context, pr
 
 // ensureProjectOwner makes sure that the owner of the project is assign to "owners" group
 func (c *projectController) ensureProjectOwner(ctx context.Context, project *kubermaticv1.Project) error {
-	var sharedOwnerPtr *kubermaticv1.User
+	var sharedOwnerPtrList []*kubermaticv1.User
 	for _, ref := range project.OwnerReferences {
 		if ref.Kind == kubermaticv1.UserKindName {
 			var sharedOwner kubermaticv1.User
 			key := types.NamespacedName{Name: ref.Name}
 			if err := c.client.Get(ctx, key, &sharedOwner); err != nil {
 				return err
 			}
-			sharedOwnerPtr = &sharedOwner
+			sharedOwnerPtrList = append(sharedOwnerPtrList, sharedOwner.DeepCopy())
 		}
 	}
-	if sharedOwnerPtr == nil {
+	if len(sharedOwnerPtrList) == 0 {
 		return fmt.Errorf("the given project %s doesn't have associated owner/user", project.Name)
 	}
-	owner := sharedOwnerPtr.DeepCopy()
 
 	var bindings kubermaticv1.UserProjectBindingList
 	if err := c.client.List(ctx, &bindings); err != nil {
 		return err
 	}
 
-	for _, binding := range bindings.Items {
-		if binding.Spec.ProjectID == project.Name && strings.EqualFold(binding.Spec.UserEmail, owner.Spec.Email) &&
-			binding.Spec.Group == GenerateActualGroupNameFor(project.Name, OwnerGroupNamePrefix) {
-			return nil
+	projectOwnerMap := map[string]bool{}
```

Comment:
Nit: since you are only looking for keys here you use a struct pointer instead 
```suggestion
	projectOwnerMap := map[string]*struct
```

Reply:
changed for hash set